test/distributed/org/apache/cassandra/fuzz/sai/SingleNodeSAITestBase.java

```diff
@@ -18,302 +18,192 @@
-                                                  SelectHelper.execute(sut, history.clock(), query),
-                                                  query);
-                    }
-                    catch (Throwable t)
-                    {
-                        logger.debug("Partition index = {}, run = {}, j = {}, i = {}", partitionIndex, run, j, i);
-
-                        Query partitionQuery = Query.selectAllColumns(schema, pd, false);
-                        QuiescentChecker.validate(schema,
-                                                  tracker,
-                                                  columns,
-                                                  reconciler.inflatePartitionState(pd, tracker, partitionQuery),
-                                                  SelectHelper.execute(sut, history.clock(), partitionQuery),
-                                                  partitionQuery);
-                        logger.debug("Partition state agrees. Throwing original error...");
-
-                        throw t;
+                        List<IdxRelation> regularRelations = HistoryBuilderHelper.generateValueRelations(rng, schema.regularColumns.size(),
+                                                                                                         column -> Math.min(schema.valueGenerators.regularPopulation(column), UNIQUE_CELL_VALUES));
+                        List<IdxRelation> staticRelations = HistoryBuilderHelper.generateValueRelations(rng, schema.staticColumns.size(),
+                                                                                                        column -> Math.min(schema.valueGenerators.staticPopulation(column), UNIQUE_CELL_VALUES));
+                        history.select(pkGen.generate(rng),
+                                       HistoryBuilderHelper.generateClusteringRelations(rng, schema.clusteringKeys.size(), ckGen).toArray(new IdxRelation[0]),
+                                       regularRelations.toArray(new IdxRelation[regularRelations.size()]),
+                                       staticRelations.toArray(new IdxRelation[staticRelations.size()]));
                     }
                 }
             }
 
-            if (run + 1 < RUNS)
-            {
-                logger.debug("Forcing compaction at the end of run {}...", run + 1);
-                compact(schema);
-            }
-        }
+            InJvmDTestVisitExecutor.replay(InJvmDTestVisitExecutor.builder()
+                                                                  .pageSizeSelector(pageSizeSelector())
+                                                                  .build(schema, history, cluster),
+                                           history);
+        });
     }
 
     protected void flush(SchemaSpec schema)
     {
         cluster.get(1).nodetool("flush", schema.keyspace, schema.table);
     }
-    
+
     protected void compact(SchemaSpec schema)
```

Comment:
`compact()` is now unused. We might need to compact every once in a while to keep the number of SSTables per partition under 32 to avoid hitting `sai_sstable_indexes_per_query_warn_threshold`

Reply:
I have added a cycled compact for now so that we could control it rather than leaving it up to auto compaction.